src/cohort-manager.js

```diff
@@ -0,0 +1,132 @@
+const Cohort = require('./cohort')
+const Student = require('./student')
+
+class CohortManager {
+  constructor() {
+    this.cohortArray = []
+    this.studentID = 1
+  }
+
+  addCohort(name, maxCapacity) {
+    if (name.length === 0) {
+      return false
+    }
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        return false
+      }
+    }
+    const newCohort = new Cohort(name, maxCapacity)
+    this.cohortArray.push(newCohort)
+    return this.cohortArray
+  }
+
+  removeCohort(name) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        this.cohortArray.splice(i, 1)
+        return this.cohortArray
+      }
+    }
+    return false
+  }
+
+  searchByCohortName(name) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (name === cohort.name) {
+        return cohort
+      }
+    }
+    return 'This cohort does not exist'
+  }
+
+  searchByStudentEmail(email) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      for (let j = 0; j < this.cohortArray[i].studentsArray.length; j++) {
+        const student = this.cohortArray[i].studentsArray[j]
+        if (email === student.email) {
+          return student
+        }
+      }
+    }
+    return 'There are no students with the entered email'
+  }
+
+  searchByStudentID(studentID) {
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      for (let j = 0; j < this.cohortArray[i].studentsArray.length; j++) {
+        const student = this.cohortArray[i].studentsArray[j]
+        if (studentID === student.studentID) {
+          return student
+        }
+      }
+    }
+    return 'There are no students with the entered email'
+  }
+
+  addStudentToCohort(cohortName, firstName, surname, githubUser, email) {
+    if (
+      this.searchByStudentEmail(email) !==
+      'There are no students with the entered email'
+    ) {
+      return false
+    }
+    const student = new Student(
+      cohortName,
+      firstName,
+      surname,
+      githubUser,
+      email
+    )
+    for (let i = 0; i < this.cohortArray.length; i++) {
+      const cohort = this.cohortArray[i]
+      if (
+        cohortName === cohort.name &&
+        cohort.studentsArray.length < cohort.maxCapacity
+      ) {
+        student.studentID = this.studentID
+        this.studentID++
+        cohort.studentsArray.push(student)
+        return student
```

Comment:
I find all your code easy to read and understand. One thing that I think would improve your work would be to split some functionality between the different classes you created. For example, if the method to add a student to a cohort belonged to the Student class it would be easier to call your searchByCohortName() here and then the call the addStudentToCohort() of that specific cohort that was returned.